src/main/kotlin/org/bitcoindevkit/karavan/Karavan.kt

```diff
@@ -31,50 +31,67 @@ class WalletController(val walletService: WalletService) {
     fun openWallet(response: HttpServletResponse, request: HttpServletRequest, @RequestBody payload: Wallet): String? {
         // stores payload of type wallet into a cookie on the client's side
         setCookie(response, payload)
-        // read cookie with key 'wallet' to confirm value
-        return readAllCookies(request)
+        return "Wallet is opened!\n"
     }
 
-    // Close wallet by dropping existing descriptor cookie by setting max-age to 0
+    // Close wallet by dropping existing descriptor cookie
     @DeleteMapping
-    fun closeWallet(request: HttpServletRequest?): String?{
+    fun closeWallet(response: HttpServletResponse, request: HttpServletRequest?): String?{
 
         val cookie = WebUtils.getCookie(request!!, "descriptor")
```

Comment:
To be safe can you loop through and timeout all the cookies in the `HttpServletRequest` instead of just the one for "descriptor" ? This would also mean you don't need to hardcode the cookie key.

Reply:
Done.